### PR TITLE
Cleanup dependency and linked service output

### PR DESCRIPTION
**Pull request by lox**
This PR uses `docker-compose up -d --scale myservice=0` to scale up service dependencies before the actual service runs, which means we can do it in a different header to avoid the noise of all the docker pull and build operations of linked containers.

The end result is ~~much~~ relatively cleaner output. The is the before:

image

This is after:

image

This also closes https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/issues/138.

### CODE REVIEW

**Comment by toolmantim** [comment]
The new output is great!

What's the new `require-prebuild` key? I've lots of questions about it, but would be great to understand the context first.

**Comment by lox** [comment]
> What's the new `require-prebuild` key? I've lots of questions about it, but would be great to understand the context first.

Damn, I was hoping you wouldn't notice that. 

It's basically just for the test suite to ensure that run steps fail if there isn't a prebuilt image. Found a whole lot of bugs and failures that were showing up green. 

I wasn't planning on documenting it 😅

**Comment by lox** [comment]
I also added a warning for when you try and `build` without an `image-repository`:

image

Might need a copy tweak now I read it again.

File: commands/build.sh [hunk 1]
```diff
@@ -12,6 +12,11 @@ service_name_cache_from_var() {
   echo "cache_from__${service_name//-/_}"
 }
 
+if [[ -z "$image_repository" ]] ; then
+  echo "+++ ⚠️ Build step doesn't have an image-repository defined"
```
Comment (toolmantim) [inline commands/build.sh]:
```suggestion
  echo "+++ ⚠️ Build step missing image-repository setting"
```

File: commands/build.sh [hunk 2]
```diff
@@ -12,6 +12,11 @@ service_name_cache_from_var() {
...
+  echo "Without a repository to push to using a prebuild doesn't do anything. The run step will build before run"
```
Comment (toolmantim) [inline commands/build.sh]:
```suggestion
  echo "This build step has no image-repository set. Without an image-repository, the Docker image won't be pushed to a repository, and won't be automatically used by any run steps."
```

**Comment by toolmantim** [comment]
Cool, if it's internal, no problem! 👌🏼

**Comment by lox** [comment]
> We should probably just deprecate `build` without an `image-repository`, if it's more often an accident than on purpose?

@toolmantim yeah, I reckon we will in the next major version. See https://github.com/buildkite-plugins/docker-compose-buildkite-plugin/issues/63

**Pull request merged**